gse-copypaste/gse-copypaste-afs/src/main/java/com/powsybl/gse/copypaste/afs/CopyManager.java

```diff
@@ -0,0 +1,226 @@
+/**
+ * Copyright (c) 2019, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.gse.copypaste.afs;
+
+import com.powsybl.afs.*;
+import com.powsybl.commons.PowsyblException;
+import com.powsybl.computation.local.LocalComputationConfig;
+import org.apache.commons.io.FileUtils;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.time.ZonedDateTime;
+import java.util.*;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.stream.Collectors;
+
+/**
+ * @author Nassirou Nambiema <nassirou.nambiena at rte-france.com>
+ */
+public final class CopyManager {
+
+    private final String storageDirectory;
+
+    private final Object copyLock = new Object();
+    private ExecutorService tPool = Executors.newFixedThreadPool(3);
+    private Map<String, CopyInfo> copy = new HashMap<>();
+
+    public CopyManager() {
+        initCopy();
+        storageDirectory = getStorageDirectory();
+    }
+
+    /**
+     * @param nodes the nodes to copy
+     */
+    public Map<String, CopyInfo> copy(List<? extends AbstractNodeBase> nodes) {
+
+        for (AbstractNodeBase node : nodes) {
+            CopyInfo copyInfo = new CopyInfo();
+            copyInfo.nodeId = node.getId();
+            copyInfo.node = node;
+            copyInfo.archivePath = Paths.get(storageDirectory + node.getName() + node.getId());
+            copyInfo.archiveComplete = false;
+            copyInfo.expirationDate = ZonedDateTime.now().plusHours(CopyServiceConstants.COPY_EXPIRATION_TIME);
+            tPool.execute(() -> {
+                // create task
+                try {
+                    archiveAndCopy(node);
+                } catch (Exception e) {
+                    throw new PowsyblException("The node " + node.getName() + " copy have encountered some problems! ");
+                }
+                // change completion status + end task
+                copyInfo.archiveComplete = true;
```

Comment:
put that after archiveAndCopy. Also you may want a 3 value boolean to indicate the state of the archive process : waiting, success, fail